modules/gdscript/gdscript_analyzer.cpp

```diff
@@ -506,6 +523,9 @@ Error GDScriptAnalyzer::resolve_class_inheritance(GDScriptParser::ClassNode *p_c
 			if (!id_type.is_set()) {
 				push_error(vformat(R"(Could not find nested type "%s".)", id->name), id);
 				return ERR_PARSE_ERROR;
+			} else if (id_type.kind != GDScriptParser::DataType::SCRIPT && id_type.kind != GDScriptParser::DataType::CLASS) {
```

Comment:
I still have some doubts about this. We should probably check if the identifier is a class or a constant containing a script/class. Also, I am not happy with the variety of error messages:

* Constant "A" is not a preloaded script or class.
* Identifier "X" is not a preloaded script or class.
* Cannot get nested types for extension from non-GDScript type "RefCounted".
* Could not find nested type "Baz".
* Cannot use variable "A" in extends chain.

Reply:
Let's fix the crash for now, and iterate on this in a follow-up.